### DISCUSSION TITLE
`MySQL server has gone away` Error

### DISCUSSION THREAD

**Discussion by TrainerGamer** [pterodactyl/panel#2980] [Support]
## :wave: Hi there!
When I try to add a database host to the panel I get this error:
`There was an error while trying to connect to the host or while executing a query: "SQLSTATE[HY000] [2006] MySQL server has gone away (SQL: SELECT 1 FROM dual)"`

The panel is running on a VPS, the daemon + MySQL server is on a dedi.
I have allowed connections to all ports of the dedi from the VPS IP thru the dedi's firewall.

Any help would be much appreciated. I've been trying to figure this out for a longgggg time.

PS. How do you use the backup system? I can't seem to find any docs regarding it.

**Comment by trenutoo** [comment]
[This ](https://laravel-news.com/laravel-and-mysql-8-fixing-mysql-server-has-gone-away-error)should resolve the issue.

Backups simply tar and move the server files into the designated backup location, you can change the location in the wings/daemon config. You can also configure files or folders to ignore in the backup process together with scheduling automated backups, in that case, the oldest backup will always be removed once the backup limit that you configured for the server is reached.

You must set the backup limit higher than 0 in the game server settings before backups can be created

**Reply by TrainerGamer** [reply]
@Software-Noob Thank you for the response :D 

`default_authentication_plugin=mysql_native_password` was actually already in my `/etc/mysql/my.cnf` file, yet I am still receiving the error. Do you have any idea what could be going on?


> Backups simply tar and move the server files into the designated backup location, you can change the location in the wings/daemon config. You can also configure files or folders to ignore in the backup process together with scheduling automated backups, in that case, the oldest backup will always be removed once the backup limit that you configured for the server is reached.
You must set the backup limit higher than 0 in the game server settings before backups can be created

Appreciate it thanks!

**Comment by julianrendell** [comment]
I just hit a similar problem. In the end it was due to the DB user for creating DBs not having permission to access the DB server.

I'm using Ubuntu Server 20.4, with MariaDB. These are the users that I needed to make it work:

```
MariaDB [(none)]> select user,host from mysql.user;
+-----------------+-----------+
| User            | Host      |
+-----------------+-----------+
| pterodactyl     | 127.0.0.1 |
| pterodactyluser | 127.0.0.1 |
| pterodactyluser | 192.%.%.% |
| mariadb.sys     | localhost |
| mysql           | localhost |
| root            | localhost |
+-----------------+-----------+
```

Notice the second entry for `pterodactyluser@192.%.%.%`. With the default docker settings (AFAIK - I don't remember changing the docker network config) the requests from the panel/wings appear be coming from somewhere on the 192.x.x.x network. I suspect it's the IP address of the "real" host. For my setup, limiting to my local network is good enough, as I may add other nodes that will need to access this db server.

(I did iterate through trying the docker IPs, but they didn't work. In frustration I tried allowing from everywhere, ie `pterodactrlyuser@%` and that worked. Then I worked backwards to a more limited setting that still worked.)

I am not sure the panel DB user (`pterodactyl@127.0.0.1`) allows access from the panel web code... but it does, so I'm not playing with that ;-)
